### PR TITLE
fix: Selling and buying amount precision fix

### PR DESCRIPTION
**Pull request by deepeshgarg007**

### CODE REVIEW

**Comment by sagarvora** [comment]
Don't we want to fetch the float precision from `System Settings` instead?

**Comment by deepeshgarg007** [comment]
@sagarvora Yes, currency_precision and float_precision should be fetched form system settings.
I have made the requested changes.

File: erpnext/accounts/report/gross_profit/gross_profit.py [hunk 1]
```diff
@@ -106,11 +106,14 @@ def process(self):
 		self.grouped = {}
 		self.grouped_data = []
 
+		currency_precision = cint(frappe.db.get_default("currency_precision")) or 3
```
Comment (Alchez) [inline erpnext/accounts/report/gross_profit/gross_profit.py]: Maybe set these as global constants in the file? You won't need to pass it as an argument.

File: erpnext/accounts/report/gross_profit/gross_profit.py [hunk 2]
```diff
@@ -349,3 +344,11 @@ def load_product_bundle(self):
 	def load_non_stock_items(self):
 		self.non_stock_items = frappe.db.sql_list("""select name from tabItem
 			where is_stock_item=0""")
+
+def set_average_rate(new_row, currency_precision, float_precision):
+		new_row.gross_profit = flt(new_row.base_amount - new_row.buying_amount, currency_precision)
```
Comment (Alchez) [inline erpnext/accounts/report/gross_profit/gross_profit.py]: This is over-indented.

**Review by Alchez — CHANGES_REQUESTED** [review]

**Review by netchampfaris — APPROVED** [review]

**Pull request merged**